### PR TITLE
is battery charing android

### PR DESCRIPTION
**Pull request by bansalayush**
## Description

Fixed issue #387 

Added `isBatteryCharging()` that allows dev to check if battery is charging or not .

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ❌     |
| Android |    ✅     |
| Windows |    ❌     |

## Checklist

* [X] I have tested this on a device/simulator for each compatible OS(android)
* [ ] I added the documentation in `README.md`.
* [ ] I mentionned this change in `CHANGELOG.md`.

### CODE REVIEW

**Comment by bansalayush** [comment]
Hey @rebeccahughes , please let me know if i need to add anything this

File: android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java [hunk 1]
```diff
@@ -211,6 +211,16 @@ public Integer getFreeDiskStorage() {
     return null;
   }
 
+  @ReactMethod
+  public void isBatteryCharging(Promise p){
+    IntentFilter ifilter = new IntentFilter(Intent.ACTION_BATTERY_CHANGED);
+    Intent batteryStatus = this.reactContext.getApplicationContext().registerReceiver(null, ifilter);
+    int status = batteryStatus.getIntExtra(BatteryManager.EXTRA_STATUS, -1);
+    boolean isCharging = status == BatteryManager.BATTERY_STATUS_CHARGING ||
+                     status == BatteryManager.BATTERY_STATUS_FULL;
```
Comment (machour) [inline android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java]: Why would isBatteryCharging() return `true` when the battery is full ? 🤔

**Comment by machour** [comment]
Hi @bansalayush and thank you for your contribution.
I left you a comment, could you have a look ?

Also, any chances you could also implement that for iOS ?

File: android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java (hunk 1 above)
Comment (bansalayush) [reply android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java]: Makes sense ... removing it

**Comment by bansalayush** [comment]
I don't know much of iOS but let me give it a try :)

**Comment by mikehardy** [comment]
This looks good and I can ask for a merge but it would need a README update (specifying it was Android only for now) and a Changelog entry

**Comment by machour** [comment]
react-native-device-info 0.27.0 published 🎉

**Pull request merged**